### ISSUE TITLE
Html.DisplayFor Enum DisplayTemplate string Model issue

### ISSUE THREAD

**Issue by michal-ciechan** [aspnet/Mvc#5948]
Why is it, that in the below case, the `Model` passed into the DisplayTemplate for an enum is of type 1string` rather than of type `ReportStatus` which is my enum type?

Report.cs
===
    public class Report
    {        
        public ReportStatus Status { get; set; }
    }

Index.cshtml
===
    @model Report
    
    @Html.DisplayFor(x => x.Status)

ReportStatus.cshtml
====
```
@model ReportStatus

@switch (Model)
{
    default:
        <div>Test</div>
        break;
}
```

Result
---
Gives the following error:

> InvalidOperationException: The model item passed into the ViewDataDictionary is of type 'System.String', but this ViewDataDictionary instance requires a model item of type 'DataTests.Models.ReportStatus'.

I can get around this for now using `@model String` in my ReportStatus.cshtml, but that means I need to use `nameof(ReportStatus.SomeStatus)` in my switch/conditionals, rather than integral conditionals.

Was this intended for whatever reason? A few scenario's right now I need a `switch` statement at the start to converting `string` back to `enum ReportStatus`.

Using AspNetCore.Mvc 1.1.2

**Comment by Eilon** [comment]
@ryanbrandenburg any thoughts on this? I think you've worked in this space a bit.

**Comment by Genbox** [comment]
We have this issue as well. A temporary workaround is to send the enum as a string to the template, then parse it back to the enum.

**Comment by dougbu** [comment]
Related to aspnet/Mvc#2430?

**Comment by ryanbrandenburg** [comment]
@michal-ciechan I've investigated this and created a reproduction app which shows this behavior for 1.1.2, and that it does not happen with the latest builds of MVC. My next step is to test some of our other releases to see if any of the rest are affected or if this is solely a 1.1.2 thing.

**Comment by ryanbrandenburg** [comment]
My investigation revealed that this scenario worked for 1.0.0, 1.0.1, 1.0.2 and 1.0.3, is broken in 1.1.0, 1.1.1 and 1.1.2, and is now working in dev.

Given that, @Eilon and I discussed this and came to the conclusion that we won't be patching 1.1.X for this scenario. If you're experiencing this issue you can either use the workaround @Genbox mentioned or use one of the unaffected versions.

**Comment by mariuszjamro** [comment]
@ryanbrandenburg Is there any plan to actually fix it?

**Comment by Eilon** [comment]
@mariuszjamro - it's fixed in 2.0.0.
